src/TaskAsCommand.php

```diff
@@ -14,55 +16,99 @@
         AsTask $taskAttribute,
         private readonly \ReflectionFunction $function,
         private readonly ContextRegistry $contextRegistry,
     ) {
-        $commandName = $taskAttribute->name;
+        $this->setDescription($taskAttribute->description);
+        $this->setAliases($taskAttribute->aliases);
 
+        $commandName = $taskAttribute->name;
         if ($taskAttribute->namespace) {
             $commandName = $taskAttribute->namespace . ':' . $commandName;
         }
 
         parent::__construct($commandName);
-
-        $this->setDescription($taskAttribute->description);
-
-        $this->setAliases($taskAttribute->aliases);
     }
 
     protected function configure(): void
     {
-        static $classToByPass = [
-            Context::class,
-            SymfonyStyle::class,
-            Application::class,
-            InputInterface::class,
-            OutputInterface::class,
-        ];
         foreach ($this->function->getParameters() as $parameter) {
-            $name = SluggerHelper::slug($parameter->getName());
-            $shortcut = null;
-            $description = '';
-            $type = $parameter->getType();
-            if (!$type instanceof \ReflectionNamedType) {
+            if (($type = $parameter->getType()) instanceof \ReflectionNamedType && \in_array($type->getName(), self::SUPPORTED_PARAMETER_TYPES, true)) {
                 continue;
             }
 
-            if (\in_array($type->getName(), $classToByPass, true)) {
+            $argAttribute = $parameter->getAttributes(AsArgument::class)[0] ?? null;
+            if ($argAttribute) {
+                /** @var AsArgument */
+                $argAttributeInstance = $argAttribute->newInstance();
+
+                $name = $this->setParameterName($parameter, $argAttributeInstance->name);
+
+                if ($parameter->isOptional()) {
+                    $mode = InputArgument::OPTIONAL;
+                } else {
+                    $mode = InputArgument::REQUIRED;
+                }
+                if (($type = $parameter->getType()) instanceof \ReflectionNamedType && 'array' === $type->getName()) {
+                    $mode |= InputArgument::IS_ARRAY;
+                }
+
+                try {
+                    $this->addArgument(
+                        $name,
+                        $mode,
+                        $argAttributeInstance->description,
+                        $parameter->isOptional() ? $parameter->getDefaultValue() : null,
+                        $argAttributeInstance->suggestedValues,
+                    );
+                } catch (LogicException $e) {
+                    throw new \LogicException(sprintf('The argument "%s" for command "%s" cannot be configured: "%s".', $parameter->getName(), $this->getName(), $e->getMessage()));
+                }
+
                 continue;
             }
+            $argAttribute = $parameter->getAttributes(AsOption::class)[0] ?? null;
```

Comment:
can we use a super class to get both option and argument attribute with the same call and when there is nothing creating a default attribute / option for this parameter ?

Then we can use the same logic for all parameters (and avoid the specific call later)

Reply:
I don't think it's possible. They are different by nature, and end use may want to use one or another one.

Then, guessing the mode for argument is quite easy, for option is much harder that I gave up! 

I'm not sure we can have the same code path for both type of argument. (of it'll be full of `if`)

Reply:
Thanks for the PR 👍🏼